multibody/kinematic/test/kinematic_evaluator_test.cc

```diff
@@ -0,0 +1,170 @@
+#include <memory>
+#include <utility>
+#include <gtest/gtest.h>
+
+#include "drake/common/test_utilities/eigen_matrix_compare.h"
+#include "drake/multibody/plant/multibody_plant.h"
+#include "drake/multibody/parsing/parser.h"
+
+#include "common/find_resource.h"
+#include "multibody/kinematic/distance_evaluator.h"
+#include "multibody/kinematic/kinematic_evaluator.h"
+#include "multibody/kinematic/world_point_evaluator.h"
+
+namespace dairlib {
+namespace multibody {
+namespace {
+
+using drake::CompareMatrices;
+using drake::multibody::MultibodyPlant;
+using drake::geometry::SceneGraph;
+using drake::multibody::Body;
+using drake::multibody::Parser;
+
+using Eigen::Vector3d;
+using Eigen::VectorXd;
+using Eigen::MatrixXd;
+
+class KinematicEvaluatorTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    plant_ = std::make_unique<MultibodyPlant<double>>(0.0);
+    auto scene_graph = std::make_unique<SceneGraph<double>>();
+    Parser parser(plant_.get(), scene_graph.get());
+    std::string full_name =
+        dairlib::FindResourceOrThrow("examples/PlanarWalker/PlanarWalker.urdf");
+
+    parser.AddModelFromFile(full_name);
+
+    plant_->WeldFrames(
+        plant_->world_frame(), plant_->GetFrameByName("base"),
+        drake::math::RigidTransform<double>());
+
+    plant_->Finalize();
+  }
+
+  std::unique_ptr<MultibodyPlant<double>> plant_;
+};
+
+TEST_F(KinematicEvaluatorTest, WorldPointEvaluatorTest) {
+  const double tolerance = 1e-10;
+
+  Vector3d pt_A({0, 0, -.5});
+  const auto& frame = plant_->GetFrameByName("right_lower_leg");
+
+  // Default evaluator: vertical normal, zero offset
+  auto evaluator = WorldPointEvaluator<double>(*plant_, pt_A, frame,
+      Vector3d({0, 0, 1}), Vector3d::Zero(), false);
+
+  auto context = plant_->CreateDefaultContext();
+
+  // Test q = 0, legs straight down
+  VectorXd q(plant_->num_positions());
```

Comment:
I remember from the past experience that it is necessary to initialize the VectorXd (for examples, to zeros) manually to avoid bad initial values